clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -247,12 +247,10 @@ private void closeInternal(final Duration timeout) {
         closeTimeout = timeout;
         wakeup();
 
-        if (timeoutMs > 0) {
-            try {
-                join(timeoutMs);
-            } catch (InterruptedException e) {
-                log.error("Interrupted while waiting for consumer network thread to complete", e);
-            }
+        try {
```

Comment:
Because one of the unit tests fails without it 😄

In the existing `KafkaConsumer.close()`, the coordinator eventually calls this method:

```java
    private void closeHeartbeatThread() {
        HeartbeatThread thread;
        synchronized (this) {
            if (heartbeatThread == null)
                return;
            heartbeatThread.close();
            thread = heartbeatThread;
            heartbeatThread = null;
        }
        try {
            thread.join();
        } catch (InterruptedException e) {
            log.warn("Interrupted while waiting for consumer heartbeat thread to close");
            throw new InterruptException(e);
        }
    }
```

By blocking via `Thread.join()`, it allows the coordinator to close fully before its underlying resources (client, etc.) are closed. In contrast, our background thread was only waiting up to the timeout, and then proceeded with closing the other resources even though the background thread may not have completed its cleanup.

If you'd prefer, I can save this change for a separate PR?

Reply:
We can keep it here. I just wanted to understand the reason behind it.